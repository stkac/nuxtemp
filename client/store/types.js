// @flow

const INCREMENT = 'increment'
const DECREMENT = 'decrement'
const FETCH_ITEM = 'fetchItem'

export {
  INCREMENT,
  DECREMENT,
  FETCH_ITEM
}
