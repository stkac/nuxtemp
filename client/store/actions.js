import * as types from './types'
import type { Item } from './state'

export default {
  incrementAsync ({ commit }) {
    setTimeout(() => {
      commit(types.INCREMENT)
    }, 200)
  },

  fetchAsync ({ commit }) {
    setTimeout(() => {
      const item: Item = {
        id: 4,
        name: 'Test item'
      }

      commit(types.FETCH_ITEM, item)
    }, 200)
  }
}
