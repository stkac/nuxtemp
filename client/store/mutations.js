// @flow

import * as types from './types'
import type { State, Item } from './state'

export default {
  [types.INCREMENT]: (state: State) => {
    state.count++
  },

  [types.DECREMENT]: (state: State) => {
    state.count--
  },

  [types.FETCH_ITEM]: (state: State, payload: Item) => {
    state.item = payload
  }
}
