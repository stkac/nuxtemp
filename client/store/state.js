// @flow

export type Item = {
  id: number,
  name: string
};

export type State = {
  count: number,
  item: ?Item
}; // without ; it breaks sublime syntax

export default function (): State {
  return {
    count: 0,
    item: null
  }
}
