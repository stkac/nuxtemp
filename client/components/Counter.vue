<template>
  <div class="counter-wrapper">
    <div class="counter">
      {{ count }}
    </div>
    <button @click="increment">Increment</button>
    <button @click="decrement">Decrement</button>
    <button @click="incrementAsync">Increment Async</button>
  </div>
</template>

<script>
import { mapState } from 'vuex'
import * as types from 'store/types'

export default {
  computed: {
    ...mapState(['count'])
  },

  methods: {
    increment () {
      this.$store.commit(types.INCREMENT)
    },
    decrement () {
      this.$store.commit(types.DECREMENT)
    },
    incrementAsync () {
      this.$store.dispatch('incrementAsync')
    }
  }
}
</script>

<style lang="scss" scoped>
.counter-wrapper {
  .counter {
    margin: 100px auto;
    border-radius: 3px;
    width: 200px;
    height: 200px;
    text-align: center;
    line-height: 200px;
    font-size: 5rem;
    background-color: #f0f0f0;
    user-select: none;
  }
}
</style>
