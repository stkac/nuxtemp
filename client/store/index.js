import Vue from 'vue'
import Vuex from 'vuex'

import createState from './state'
import mutations from './mutations'
import actions from './actions'

Vue.use(Vuex)

const strict = process.env.NODE_ENV === 'production'

export default function () {
  return new Vuex.Store({
    state: createState(),
    strict,
    mutations,
    actions
  })
}
