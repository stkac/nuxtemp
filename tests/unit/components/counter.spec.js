import Vuex from 'vuex'
import { mount, createLocalVue } from 'vue-test-utils'
import Counter from 'components/Counter'
import createStore from 'store'
import * as types from 'store/types'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Counter component unit tests', () => {
  let store

  beforeEach(() => {
    store = createStore()
  })

  it('should have three buttons', () => {
    const wrapper = mount(Counter, { store, localVue })

    expect(wrapper.findAll('button').length).toBe(3)
  })

  it('should have 0 as a default counter value', () => {
    const wrapper = mount(Counter, { store, localVue })

    expect(wrapper.find('.counter').text().trim()).toBe('0')
  })

  it('should increment counter', () => {
    const wrapper = mount(Counter, { store, localVue })

    wrapper.vm.$store.commit(types.INCREMENT)
    expect(wrapper.vm.count).toBe(1)
  })

  it('should decrement counter', () => {
    const wrapper = mount(Counter, { store, localVue })

    wrapper.vm.$store.commit(types.DECREMENT)
    expect(wrapper.vm.count).toBe(-1)
  })

  it('should match the snapshot', () => {
    const wrapper = mount(Counter, { store, localVue })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
