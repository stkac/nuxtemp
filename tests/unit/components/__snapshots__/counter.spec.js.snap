// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component unit tests should match the snapshot 1`] = `
"<div class=\\"counter-wrapper\\"><div class=\\"counter\\">
    0
  </div> <button>Increment</button> <button>Decrement</button> <button>Increment Async</button></div>"
`;
